### PR TITLE
Improve SQL routine description

### PR DESCRIPTION
**Pull request by mosabua**
## Description

I hope to work against the assumption of some readers that routines can invoke other SQL statements and are more than simple functions with this addition. The Limitation section already details that but maybe this can help some more. 

## Additional context and related issues

There are also suggestions to say that a function operates from data from one row, but I think that is actually wrong. For example you could write a JOIN that combines multiple rows and then uses values from the different rows as input value for a function.

I was thinking about using "scalar function" .. but I am not 100% sure.. since I think it could return a JSON construct. Also .. not sure if readers necessary understand without defining it more.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Review by Ordinant — APPROVED** [review]
LGTM. If approved by others, I'll use this wording in the Galaxy pages.

**Review by wendigo — APPROVED** [review]

File: docs/src/main/sphinx/routines.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # SQL routines
 
 A SQL routine is a custom, user-defined function authored by a user of Trino in
-a client and written in SQL. More details are available in the following sections:
+a client and written in SQL. Routines are functions that process one or more
```
Comment (electrum) [inline docs/src/main/sphinx/routines.md:4]: I think this is worse than before. The "one or more input values" makes it sound like an aggregation function. How about

> Routines are scalar functions that return a single output value.

The Trino SPI has `FunctionKind` with `SCALAR` which is what these functions are.

Comment (electrum) [reply docs/src/main/sphinx/routines.md:4]: If you think of the input arguments as a tuple, then the input is actually a single value. Note that the output type can also be a tuple.

Comment (mosabua) [reply docs/src/main/sphinx/routines.md:4]: ok .. lets go with the scalar version ..

Comment (mosabua) [reply docs/src/main/sphinx/routines.md:4]: Updated now

**Comment by mosabua** [comment]
@electrum did some more refinement .. I hope this is ok now.

File: docs/src/main/sphinx/routines/introduction.md [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
 
 A SQL routine is a custom, user-defined function authored by a user of Trino and
-written in the SQL routine language. You can declare the routine body within a
-[](/routines/function) block as [inline routines](routine-inline) or [catalog
-routines](routine-catalog).
+written in the SQL routine language. Routines are scalar functions that return a
+single output value, similar to [built-in functions](/functions).
+
+[Declare the routine](routine-declaration) with a `FUNCTION` definition using
+the supported SQL routine statements. A routine can be declared and used as
```
Comment (electrum) [inline docs/src/main/sphinx/routines/introduction.md:8]: "used as an"

File: docs/src/main/sphinx/routines/introduction.md [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+[inline routine](routine-inline) or declared as [catalog routine](routine-catalog)
```
Comment (electrum) [inline docs/src/main/sphinx/routines/introduction.md:9]: "as a"

**Review by electrum — APPROVED** [review]

**Comment by mosabua** [comment]
Fyi @Jessie212 .. this will force a rebase on your PR probably

**Pull request merged**